airflow/providers/databricks/sensors/databricks.py

```diff
@@ -0,0 +1,103 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Databricks sensors"""
+
+from typing import TYPE_CHECKING, Any, Dict, Optional, Sequence, Union
+
+from airflow.exceptions import AirflowException
+from airflow.providers.databricks.hooks.databricks import DatabricksHook
+from airflow.sensors.base import BaseSensorOperator
+from airflow.utils.context import Context
+
+if TYPE_CHECKING:
+    from airflow.sensors.base import PokeReturnValue
+
+
+class DatabricksJobRunSensor(BaseSensorOperator):
+    """
+    Check for the state of a submitted Databricks job run or specific task of a job run.
+
+    :param run_id: Id of the submitted Databricks job run or specific task of a job run. (templated)
```

Comment:
It would make sense to refer to a job by name or ID, and a time frame… Run ID is very specific to a specific execution…

it would make sense to list use cases for the sensor. For example, check for a specific job (by name or ID) executed today/yesterday/…

Reply:
I agree. At least we should be able to check for a job id. But a job might have multiple job runs. Maybe we can wait for the latest active job run. It could be like this:

```
:param job_id: Id of the Databricks Job. Since a job might have multiple runs, this sensor waits for the most recent job run that is not terminated.
```

It means, when this sensor is fired, it lists the job runs using the [JobsRunsList API](https://docs.databricks.com/dev-tools/api/latest/jobs.html#operation/JobsRunsList) with passing ``active_only=True``. If there is one or many active tasks, it waits for the most recent one.

How's that sound?

Reply:
Also, we can check for a job name, or a task name. But, as far as i know, names are not distinct. So, there might be multiple jobs with the same name. How should we handle that?

Reply:
Job names aren’t distinct, but it could be still handled as its done in RunNow operator…

task names inside job are distinct

Reply:
Isn't the base use-case for this that you've started a job (or jobs) by submitting it in a previous task and now you want to wait for their completion?

Then other use-cases like looking up a job by name, those seem a bit more esoteric to me. Perhaps that can be done using Python code if we provide a code example?

Sometimes less is more.

Reply:
Not necessary to wait for the job in current pipeline. it could be to wait for results from jobs in other pipelines, etc.

Reply:
Could be – but it's less obvious that this situation is a one-size-fits-all. Maybe in this case you do need some Python logic regardless to determine what to do in case multiple jobs are active, etc.